### PR TITLE
Subscribe with offset

### PR DESCRIPTION
**Pull request by cyx**

### CODE REVIEW

File: broker/broker.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 
 type Broker interface {
 	io.Writer
-	Subscribe() (chan []byte, error)
+	Subscribe(offset int64) (chan []byte, error)
```
Comment (tt) [inline broker/broker.go]: Unless a stream can really be more than ~2 GiB, `int32`?

File: broker/redis.go [hunk 2]
```diff
@@ -117,6 +118,7 @@ func (b *RedisBroker) Subscribe() (ch chan []byte, err error) {
 	defer b.mutex.Unlock()
 	ch = make(chan []byte, msgBuf)
 	b.subscribers[ch] = true
+	b.position = offset
```
Comment (tt) [inline broker/redis.go]: A broker might have multiple subscribers, so this is not safe.

File: broker/broker.go (hunk 1 above)
Comment (cyx) [inline broker/broker.go]: yeah, was mostly playing with the existing position type, but if you think it's safe, yes can change that.

**Comment by tt** [comment]
Looks good to me.

**Pull request merged**